client/executor/src/native_executor.rs

```diff
@@ -269,12 +281,108 @@ impl<D: NativeExecutionDispatch> RuntimeInfo for NativeExecutor<D> {
 			runtime_code,
 			ext,
 			false,
-			|_instance, version, _ext|
+			|_module, _instance, version, _ext|
 				Ok(version.cloned().ok_or_else(|| Error::ApiError("Unknown version".into()))),
 		)
 	}
 }
 
+pub struct RuntimeInstanceSpawn {
+	module: Arc<dyn sc_executor_common::wasm_runtime::WasmModule>,
```

Comment:
Isn't `WasmModule` already available in the scope? Why won't we use that?

Also applies to the other instances below.

Reply:
I'd avoid accessing thread state from random locations around the codebase

Reply:
I mean, why do use fully-qualified name instead of just `WasmModule`?